backend/FwLite/FwDataMiniLcmBridge/Api/FwDataMiniLcmApi.cs

```diff
@@ -742,20 +742,45 @@
         return await GetEntry(entry.Id) ?? throw new InvalidOperationException("Entry was not created");
     }
 
-    public Task<ComplexFormComponent> CreateComplexFormComponent(ComplexFormComponent complexFormComponent)
+    public Task<ComplexFormComponent> CreateComplexFormComponent(ComplexFormComponent complexFormComponent, BetweenPosition<ComplexFormComponent>? position = null)
     {
         UndoableUnitOfWorkHelper.DoUsingNewOrCurrentUOW("Create Complex Form Component",
             "Remove Complex Form Component",
             Cache.ServiceLocator.ActionHandler,
             () =>
             {
                 var lexEntry = EntriesRepository.GetObject(complexFormComponent.ComplexFormEntryId);
-                AddComplexFormComponent(lexEntry, complexFormComponent);
+                AddComplexFormComponent(lexEntry, complexFormComponent, position);
             });
         return Task.FromResult(ToComplexFormComponents(EntriesRepository.GetObject(complexFormComponent.ComplexFormEntryId))
             .Single(c => c.ComponentEntryId == complexFormComponent.ComponentEntryId && c.ComponentSenseId == complexFormComponent.ComponentSenseId));
     }
 
+    public Task MoveComplexFormComponent(Guid complexFormEntryId, ComplexFormComponent complexFormComponent, BetweenPosition<ComplexFormComponent> between)
+    {
+        if (!EntriesRepository.TryGetObject(complexFormEntryId, out var lexComplexFormEntry))
+            throw new InvalidOperationException("Entry not found");
+
+        var complexFormComponentEntityId = complexFormComponent.ComponentSenseId ?? complexFormComponent.ComponentEntryId;
+        var lexComponent = FindSenseOrEntryComponent(complexFormComponentEntityId);
```

Comment:
it doesn't look like `complexFormComponentEntityId` is being used other than here, so you could change `FindSenseOrEntryComponent` to take a `ComplexFormComponent`. Once you do that you can also use `FindSenseOrEntryComponent` from `internal AddComplexFormComponent` below.